provision/acc_provision/acc_provision.py

```diff
@@ -225,6 +225,8 @@ def config_default():
             "kubectl": "kubectl",
             "system_namespace": "aci-containers-system",
             "ovs_memory_limit": "1Gi",
+            "containers_controller_memory_limit": "3Gi",
+            "containers_host_memory_limit": "3Gi",
```

Comment:
Can we provide container specific resource limits options?? also we can remove containers_ prefix in names.

Reply:
That would become too granular and will end up showing many such options in kube_confg

Reply:
memory usage of mcast-daemon container is less as compared to aci-containers-host opflex-agent , so i think better to provide container specific options, what do you think @snaiksat ??

Reply:
We are setting memory limit not request. Limit will not allocate 3Gi, it will make sure containers does acquire more than this.

Reply:
By default requests memory is set same as limits.

name: aci-containers-openvswitch
resources:
limits:
memory: 2Gi
requests:
memory: 2Gi

Reply:
If we set only "limit" then k8s sets "request" value same as "limit". 
https://kubernetes.io/docs/tasks/administer-cluster/manage-resources/memory-default-namespace/#what-if-you-specify-a-container-s-limit-but-not-its-request
That answers why we see this behaviour. Along with limit we need to set request as well.

Reply:
Added LimitRange namespaced resource to set aci-containers pod's default memory request and limit to 128Mi and 3Gi respectively. Haven't changed existing limit parameter of openvswitch pod.

Reply:
We can remove config param (https://github.com/noironetworks/acc-provision/blob/85d03ea182aebe188a6ae8bec2eb2a0132a8c67a/provision/acc_provision/templates/provision-config.yaml#L101) to avoid conflict with newer namespace wide changes.

Reply:
i thought about it earlier but i think we can leave it, (a) for backward compat and (b) also providing the option to fine tune OVS limits

Reply:
With this we will be setting 2 limit 1Gi(pod specific) and 3 Gi(ns wide limit), not sure if that will work.

Reply:
@PariyaAshok Pod specific values will take precedence over namespace specific values.
Okay @snaiksat, so will not touch existing ovs memory limit references for backward compatibility

Reply:
@snaiksat  With namespace wide limit set, we see following, With that we are allowing max memory of aci-containers-host pod to 9Gi and operator pod to 6Gi.

Age
  ---------                                    ----                                                          ------------  ----------  ---------------  -------------  ---
  aci-containers-system                        aci-containers-controller-859b879c94-7bw88                    0 (0%)        0 (0%)      128Mi (0%)       3Gi (20%)      2m40s
  aci-containers-system                        aci-containers-host-rg692                                     0 (0%)        0 (0%)      384Mi (2%)       9Gi (61%)      24s
  aci-containers-system                        aci-containers-openvswitch-4cqwd                              0 (0%)        0 (0%)      1Gi (6%)         1Gi (6%)       24s
  aci-containers-system                        aci-containers-operator-7c6b5d6f84-66m7m                      0 (0%)        0 (0%)      256Mi (1%)       6Gi (41%)      3m19s